frontera/contrib/backends/cassandra/models.py

```diff
@@ -0,0 +1,101 @@
+# -*- coding: utf-8 -*-
+import uuid
+from cassandra.cqlengine.models import Model
+from cassandra.cqlengine.usertype import UserType
+from cassandra.cqlengine.columns import Map, Text, Float, Integer, DateTime, UserDefinedType, Counter, Boolean, \
+    SmallInt, BigInt
+
+
+class Meta(UserType):
+    domain = Map(Text(), Text(), required=False)
+    fingerprint = Text()
+    origin_is_frontier = Boolean()
+    scrapy_callback = Text()
+    scrapy_errback = Text()
+    scrapy_meta = Map(Text(), Text(), required=False)
+    score = Float(required=False)
+    jid = Integer(required=False)
+
+
+class MetadataModel(Model):
+    __table_name__ = 'metadata'
+
+    crawl = Text(primary_key=True)
+    fingerprint = Text(primary_key=True)
+    url = Text(index=True)
+    depth = Integer()
+    created_at = DateTime()
+    fetched_at = DateTime(required=False)
+    status_code = Integer(required=False)
+    score = Float(required=False)
+    error = Text(required=False)
+    meta = UserDefinedType(Meta)
+    headers = Map(Text(), Text(), required=False)
+    cookies = Map(Text(), Text(), required=False)
+    method = Text(required=False)
+
+    @classmethod
+    def query(cls, session):
+        return session.query(cls)
+
+    def __repr__(self):
+        return '<Metadata:%s (%s)>' % (self.url, self.fingerprint)
+
+
+class StateModel(Model):
+    __table_name__ = 'states'
+
+    crawl = Text(primary_key=True)
+    fingerprint = Text(primary_key=True)
+    state = SmallInt(index=True)
+
+    @classmethod
+    def query(cls, session):
+        return session.query(cls)
+
+    def __repr__(self):
+        return '<State:%s=%d>' % (self.fingerprint, self.state)
+
+
+class QueueModel(Model):
+    __table_name__ = 'queue'
+
+    crawl = Text(primary_key=True)
```

Comment:
Yes, you've got my question right.
Just a few remarks:
- sort column shouldn't be connected with run mode. Sorting by score can be used in single process mode, and by creation date (FIFO) in distributed. It depends on application goals.
- fingerprint is needed by Frontera to operate, but it's wrong to use it for deletion of items in the queue (e.g in `get_next_results`). The same fingerprint can be scheduled few times, and it's not a mistake, it can be required by the application. Auto incrementing field, is a possible solution.

I have a big doubts that someone will be using Frontera with Cassandra in non-distributed mode. Cassandra is designed for big volumes of data and it's distributed, so I would limit support of single process mode to minimum. Mostly for debugging and testing purposes. Therefore, IMO sorting only by score, should be enough.